file/src/test/java/docs/javadsl/ArchiveTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package docs.javadsl;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.japi.Pair;
+import akka.stream.ActorMaterializer;
+import akka.stream.IOResult;
+import akka.stream.Materializer;
+import akka.stream.alpakka.file.ArchiveMetadata;
+import akka.stream.alpakka.file.javadsl.Archive;
+import akka.stream.javadsl.FileIO;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.stream.testkit.javadsl.StreamTestKit;
+import akka.testkit.TestKit;
+import akka.util.ByteString;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+public class ArchiveTest {
+  private ActorSystem system;
+  private Materializer mat;
+  private ArchiveHelper archiveHelper;
+
+  @Before
+  public void setup() throws Exception {
+    system = ActorSystem.create();
+    mat = ActorMaterializer.create(system);
+    archiveHelper = new ArchiveHelper();
+  }
+
+  @Test
+  public void flowShouldCreateZIPArchive() throws Exception {
+    Path filePath1 = getFileFromResource("akka_full_color.svg");
+    Path filePath2 = getFileFromResource("akka_icon_reverse.svg");
+
+    Source<ByteString, NotUsed> source1 =
+        FileIO.fromPath(filePath1).mapMaterializedValue((v) -> NotUsed.getInstance());
+    Source<ByteString, NotUsed> source2 =
+        FileIO.fromPath(filePath2).mapMaterializedValue((v) -> NotUsed.getInstance());
+
+    /*
+    // #sample
+    Source<ByteString, NotUsed> source1 = ...
+    Source<ByteString, NotUsed> source2 = ...
+    // #sample
+    */
+
+    // #sample
+    Pair<ArchiveMetadata, Source<ByteString, NotUsed>> pair1 =
+        Pair.apply(ArchiveMetadata.create("akka_full_color.svg"), source1);
+    Pair<ArchiveMetadata, Source<ByteString, NotUsed>> pair2 =
+        Pair.apply(ArchiveMetadata.create("akka_icon_reverse.svg"), source2);
+
+    Source<Pair<ArchiveMetadata, Source<ByteString, NotUsed>>, NotUsed> source =
+        Source.from(Arrays.asList(pair1, pair2));
+
+    Sink<ByteString, CompletionStage<IOResult>> fileSink = FileIO.toPath(Paths.get("logo.zip"));
+    CompletionStage<IOResult> ioResult = source.via(Archive.zipFlow()).runWith(fileSink, mat);
+
+    // #sample
+
+    ioResult.toCompletableFuture().get(3, TimeUnit.SECONDS);
+
+    archiveHelper.createReferenceZipFile(Arrays.asList(filePath1, filePath2), "logo-reference.zip");
+
+    final Sink<ByteString, CompletionStage<ByteString>> foldSink =
+        Sink.fold(ByteString.empty(), ByteString::concat);
```

Comment:
`ByteString.empty()` should be replaced to `ByteString.emptyByteString()` so it compiles with Scala 2.13.